### PR TITLE
Added check for existing recommendation in friends route. Fixes #20

### LINKED ISSUE #20: Change friends route to accept an optional item id
Given an item, we need to grey out those friends who already have that item recommended.

### PR DESCRIPTION
**Pull request by aandis**

### CODE REVIEW

File: app/controllers/api/friends_controller.rb [hunk 1]
```diff
@@ -5,23 +5,33 @@ def index
...
+    if params[:item_type].present? ^ params[:item_id].present?
+      render plain: "Provide a pair of parameters, item_id and item_type", status: 400 and return
     end
 
-    # If has_item_type exists, should be valid
-    unless [nil, 'AndroidApp'].include?(params[:has_item_type])
-      render plain: "Invalid has_item_type, should be AndroidApp", status: 400 and return
+    # If item_type exists, should be valid
+    unless [nil, 'Movie', 'AndroidApp'].include?(params[:item_type])
+      render plain: "Invalid item_type, should be AndroidApp", status: 422 and return
```
Comment (rohitpaulk) [inline app/controllers/api/friends_controller.rb]: `422` eh? Fancy.

File: app/controllers/api/friends_controller.rb [hunk 2]
```diff
@@ -5,23 +5,33 @@ def index
...
 
-    if params[:has_item_type] && params[:has_item_id]
-      result = result.each.map do |user|
-        has_item = !user.android_apps.where(id: params[:has_item_id]).empty?
-        user = user.serializable_hash
-        user[:has_item] = has_item
-        user
+    if params[:item_type] && params[:item_id]
+      result = result.each.map do |follower|
+        has_item = !follower.user_items.where(
```
Comment (rohitpaulk) [inline app/controllers/api/friends_controller.rb]: Can we change this to `x.where(...).exists?` instead of `!x.where(...).empty?`?

**Comment by rohitpaulk** [comment]
This looks good to me, man. Can you update the tests? Should be pretty straightforward.

File: app/controllers/api/friends_controller.rb (hunk 1 above)
Comment (aandis) [inline app/controllers/api/friends_controller.rb]: :sunglasses:

**Comment by aandis** [comment]
heroku push whatever you merge,

**Pull request merged**